### PR TITLE
ARISTOTLE: use the `preferred` shakemap version if more than one is available

### PR DESCRIPTION
**Pull request by ptormene**
I implemented a logic that follows the description of `preferredWeight` in https://earthquake.usgs.gov/data/comcat/#product_preferredWeight, i.e.:
`Relative weight of product. When multiple products of the same type are associated to an event, one product of each type is considered "most preferred" This is defined as the product of that type with the largest preferredWeight, and when two products have equal preferredWeight the most recent updateTime is more preferred.`

I also tried to collect some information about the shakemap that is used (see https://github.com/gem/oq-engine/commit/bd96e944b4e7c34aad37026175c90ad35e9cefe8), but it is not so straightforward to propagate such information in order to make it visible in the outputs page, so I am postponing that work to a separate PR.

### CODE REVIEW

File: openquake/hazardlib/shakemap/parsers.py [hunk 1]
```diff
@@ -441,64 +455,64 @@ def download_station_data_file(usgs_id, save_to_home=False):
...
+    shakemap = get_preferred_shakemap(shakemaps)
+    contents = shakemap['contents']
+    if 'download/stationlist.json' in contents:
+        stationlist_url = contents.get('download/stationlist.json')['url']
+        logging.info('Downloading stationlist.json')
+        stations_json_str = urlopen(stationlist_url).read()
+        try:
+            stations = read_usgs_stations_json(stations_json_str)
+        except (LookupError, UnicodeDecodeError, JSONDecodeError) as exc:
+            logging.info(str(exc))
```
Comment (micheles) [inline openquake/hazardlib/shakemap/parsers.py:472]: I would have expected a `logging.error` here, not `logging.info`.

Comment (ptormene) [reply openquake/hazardlib/shakemap/parsers.py]: I changed that and some other logging types

File: openquake/hazardlib/shakemap/parsers.py [hunk 2]
```diff
@@ -441,64 +455,64 @@ def download_station_data_file(usgs_id, save_to_home=False):
...
-                    station_data_file = temp_file.name
-                    df.to_csv(station_data_file, encoding='utf8', index=False)
-                    logging.info(f'Wrote stations to {station_data_file}')
-                    return station_data_file
+                msg = 'No stations were found'
+                logging.info(msg)
+                raise LookupError(msg)
+        else:
+            with tempfile.NamedTemporaryFile(
+                    delete=False, mode='w+', newline='',
```
Comment (micheles) [inline openquake/hazardlib/shakemap/parsers.py:510]: You should use `general.gettemp`, which honors the custom_tmp

File: openquake/hazardlib/shakemap/parsers.py [hunk 3]
```diff
@@ -441,64 +455,64 @@ def download_station_data_file(usgs_id, save_to_home=False):
...
+            seismic_len = len(
+                stations[stations['station_type'] == 'seismic'])
+        except KeyError:
+            msg = (f'{original_len} stations were found, but the'
+                   f' "station_type" is not specified, so we can not'
+                   f' identify the "seismic" stations.')
+            logging.info(msg)
+            raise LookupError(msg)
+        df = usgs_to_ecd_format(stations, exclude_imts=('SA(3.0)',))
+        if save_to_home:
```
Comment (micheles) [inline openquake/hazardlib/shakemap/parsers.py:485]: Why do we want to save in the home directory?

Comment (ptormene) [reply openquake/hazardlib/shakemap/parsers.py:485]: It is explained in the docstring. I wanted to have a simple way to check the contents of the station data before and after the conversion, for debugging purposes. It is disabled by default.

File: openquake/hazardlib/shakemap/parsers.py (hunk 2 above)
Comment (ptormene) [reply openquake/hazardlib/shakemap/parsers.py]: Done

**Review by micheles — APPROVED** [review]

**Pull request merged**